codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -161,29 +161,17 @@ public B gracefulShutdownTimeoutMillis(long gracefulShutdownTimeoutMillis) {
         }
 
         /**
-         * Determine if the encoder should queue frames to honor the value set by
-         * {@link #encoderMaxConcurrentStreams(int)}.
-         */
-        public B encoderEnforceMaxConcurrentStreams(boolean encoderEnforceMaxConcurrentStreams) {
-            encoderMaxConcurrentStreams = -1;
-            return thisB();
-        }
-
-        private boolean encoderEnforceMaxConcurrentStreams() {
-            return encoderMaxConcurrentStreams >= 0;
-        }
-
-        /**
          * How many initial streams are allowed to exists concurrently. Frames will be queued if they would result in
          * creating a stream which would cause the number of existing streams to exceed this number.
-         * @see #encoderEnforceMaxConcurrentStreams(boolean)
+         *
+         * Use {@code -1} if you not want to enforce max concurrent streams.
          */
         public B encoderMaxConcurrentStreams(int encoderMaxConcurrentStreams) {
             // This bounds are enforced here because the builder makes assumptions about its valid range to determine
             // if it should be used.
-            if (encoderMaxConcurrentStreams < 0) {
+            if (encoderMaxConcurrentStreams < -1) {
```

Comment:
@normanmaurer - No `int`, just a `boolean` to control weather to use `StreamBufferingEncoder`. For the client this should default to `true` and for the server it should default to false. For the server if it is true at build time, instead of creating `StreamBufferingEncoder` we should throw an error (`StreamBufferingEncoder` is not supported on the server yet https://github.com/netty/netty/issues/4346).